docs/ParameterValidation.md

```diff
@@ -0,0 +1,176 @@
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid year parameter
+
+- Name: rating
+- Type: double
+- Parameter Validation:
+  - Valid input range: [0.0, 10.0]
+  - Round off more than single decimal to a single decimal
+    - Example: round off 7.253896 to 7.3
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by rating as [{}, {},..], empty array when no results, where movie.rating >= rating
+  - Invalid input: input that does not parse or is out of range
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid rating parameter
+
+- Name: actorId
+- Type: string
+- Parameter Validation
+  - Valid input : well formed actorId rule - starts with lower case 'nm' followed by 5-9 digits, parsed as an int. This is case sensitive input.
+    - Example: 'nm1265067'
+    - Status : 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by actorid as [{}, {},..], empty array when no results, where actorId was in movie
+  - Invalid input: values not conforming to well formed actorid rule.  
+    - Example: 'nm1234', 'ab1234'
+    - Status : 400
+    - Content-Type: text/plain
+    - Response Body: Invalid actorid parameter
+
+- Name: genre
+- Type: string
+- Parameter Validation:
+  - Valid input: case insensitive genre , length [3, 20]
+    - Example: 'War','war','Documentary', 'documentary'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by genre as [{}, {},..], empty array when no results,  where movie.genres contains genre
+  - Invalid input: genre length < 3 or >20 characters
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid genre parameter
+
+- Name: pageSize
+- Type: integer
+- Description: pageSize limits the number of results returned from the server to a specified number of results
+- Parameter Validation:
+  - Default value: 100
+  - Valid input range: [1, 1000]
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies limited to pageSize as [{}, {},..], empty array when no results
+  - Invalid input: input that does not parse or is out of range
+    - Example: 100.23
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid pageSize parameter
+
+- Name: pageNumber
+- Type: integer
+- Description: pageNumber sets the first position to return from the results of the query
+- Parameter Validation:
+  - Default value: 1
+  - Valid input range: [1, 10000]
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies based on pageNumber as [{}, {},..], empty array when no results
+  - Invalid input: input that does not parse or is out of range
+    - Example: 100.23
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid pageNumber parameter
+
+### API Parameters
+
```

Comment:
Name this header: "Movies API Parameters" to be consistent with Actors header below